### DISCUSSION TITLE
Cannot playback recording.mavlink after building MavLinkTest

### DISCUSSION THREAD

**Discussion by Telomeresis** [microsoft/AirSim#4314] [Support Q&A]
Hi, I am unable to playback the following file: https://microsoft.github.io/AirSim/logs/recording.mavlink using AirSim on Windows. The steps I'm following is from this link: https://microsoft.github.io/AirSim/playback/

This is what I've done so far:
1. I built MavLinkTest using VS 2019 in Debug mode
2. I started a server using the following command: `MavLinkTest -server:127.0.0.1:14550 -udp` (the steps do not reflect the need for the udp/tcp/serial argument)
3. This occurs: 
```
isLocalControlSupported failed: Two second timeout waiting for response to mavlink command MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES
isLocalControlSupported failed: Two second timeout waiting for response to mavlink command MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES
isLocalControlSupported failed: Two second timeout waiting for response to mavlink command MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES
isLocalControlSupported failed: Two second timeout waiting for response to mavlink command MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES
isLocalControlSupported failed: Two second timeout waiting for response to mavlink command MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES
Downloading drone parameters so we know how to control it properly...
Ready...
```
4. I ran the following command: `PlayLog recording.mavlink` (However, it is not recognized).
5. After typing '?' to get the list of commands, I then used this command: 'playlog recording.mavlink' (However, it still doesn't work)

Is there some issue with the recording.mavlink file, or perhaps must the file be located in a very specific directory? Can anyone who has gotten the log playback functionality working shed some light on this issue?

**Comment by zimmy87** [comment]
The syntax of MavLinkTest's commandline arguments changed with #3386, so step 2 should be changed to `MavLinkTest -server -udp:127.0.0.1:14550`. Also, what settings are you using to start AirSim? You'll need to configure AirSim to connect to PX4 (like described [here](https://microsoft.github.io/AirSim/px4_sitl/)) in order to accept a MavLink connection. I tried setting this up with my local PX4 repo in WSL2, but MavLinkTest doesn't seem to be able to currently host a server on the WSL ethernet adapter, so I wasn't able to get a successful connection myself. If I have time, I'll try to get this working with Cygwin instead, but hopefully this can at least get you pointed in the right direction.
